guava/src/com/google/common/eventbus/EventBus.java

```diff
@@ -104,6 +104,86 @@
   private final SubscriberRegistry subscribers = new SubscriberRegistry(this);
   private final Dispatcher dispatcher;
 
+  /**
+   * A builder for creating instances of {@link EventBus}. Not reusable, if attempted to be used more
+   * than once, throws {@link IllegalStateException} Example: <pre> {@code
+   *   EventBus bus = new EventBus.Builder()
+   *                  .identifier("EVENT_BUS")
+   *                  .executor(MoreExecutors.directExecutor())
+   *                  .dispatcher(Dispatcher.immediate())
+   *                  .exceptionHandler(new CustomSubscriberExceptionHandler())
+   *                  .build();}</pre>
+   */
+  public static class Builder {
```

Comment:
I suggest declaring this class `final`, adding a private constructor, and adding a `builder()` static method in `EventBus.class` for users to access the builder. Should be similar to [the builder for immutable lists](https://github.com/google/guava/blob/2d0957c2bf1ee888f02119e84d824be8393c6145/guava/src/com/google/common/collect/ImmutableMap.java#L183).